src/js_sync.hpp

```diff
@@ -1100,6 +1108,22 @@ void SyncClass<T>::populate_sync_config(ContextType ctx, ObjectType realm_constr
         config.sync_config->error_handler = std::move(error_handler);
         config.sync_config->is_partial = is_partial;
 
+        SyncSessionStopPolicy session_stop_policy = SyncSessionStopPolicy::AfterChangesUploaded;
+        ValueType session_stop_policy_value = Object::get_property(ctx, sync_config_object, "_sessionStopPolicy");
+        if (!Value::is_undefined(ctx, session_stop_policy_value)) {
+            std::string stop_session = Value::validated_to_string(ctx, session_stop_policy_value, "_sessionStopPolicy");
+            if (stop_session == std::string("immediately")) {
```

Comment:
Not a C++ expert, but wouldn't this be more naturally written as a `switch` rather than a bunch of if-s?

Reply:
because in C++ you can't use switch on strings  :-)

Reply:
But I don't get why a string enum is used to store a config value instead of a numeric enum, which would remove a number of failure options?

Reply:
JS doesn't support enums, so historically in the JS API enum arguments are always parsed in as strings. You could argue we could create a "type-safe" enum object in JS, but it would only really help with auto-complete in editors. The compiler would still let you do awful things.

Personally, I think having enum objects in JS would be nice, but that would probably be something we need to consider as a general API enhancement. Note, that TypeScript do have an enum definition.